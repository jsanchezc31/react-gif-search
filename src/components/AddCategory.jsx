import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("OnePunch");

  // Para actualizar el input text ya que sino no funciona en react
  const onInputChange = (e) => {
    // console.log(e.target.value);
    setInputValue(e.target.value);
  };

  // Al precionar enter se aÃ±ade un nuevo valor a la lista
  const onSubmit = (e) => {
    e.preventDefault();
    // console.log(inputValue);
    if (inputValue.trim().length <= 1) return;

    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form
      onSubmit={(e) => {
        onSubmit(e);
      }}
      aria-label="form"
    >
      <input
        type="text"
        placeholder="Buscar..."
        value={inputValue}
        // onChange={(e) => onInputChange(e)}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
