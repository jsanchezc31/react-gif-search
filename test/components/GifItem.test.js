import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas de Gif Item", () => {
  const title = "Titulo prueba";
  const url = "https://test-url/";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con la URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);

    // expect(screen.getByRole("img").alt).toBe(url);
    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("debe de mostrar el titulo en la imagen", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
