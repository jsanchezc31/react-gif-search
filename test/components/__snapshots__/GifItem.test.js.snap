// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Gif Item Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo prueba"
      src="https://test-url/"
    />
    <p>
      Titulo prueba
    </p>
  </div>
</div>
`;
